core/src/avm2/globals/toplevel.rs

```diff
@@ -379,3 +379,95 @@ fn encode_utf8_with_exclusions<'gc>(
 
     Ok(AvmString::new_utf8(activation.context.gc_context, output).into())
 }
+
+pub fn decode_uri<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    _this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    decode(
+        activation,
+        args,
+        // Characters that are reserved, sourced from as3 docs
+        "#$&+,/:;=?@",
+        "decodeURI",
+    )
+}
+
+pub fn decode_uri_component<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    _this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    decode(activation, args, "", "decodeURIComponent")
+}
+
+// code derived from flash.utils.unescapeMultiByte
+fn decode<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    args: &[Value<'gc>],
+    reserved_set: &str,
+    func_name: &str,
+) -> Result<Value<'gc>, Error<'gc>> {
+    let value = match args.first() {
+        None => return Ok("undefined".into()),
+        Some(Value::Undefined) => return Ok("null".into()),
+        Some(value) => value.coerce_to_string(activation)?,
+    };
+
+    //FIXME: support bugzilla #538107 and #609416
+    let mut output = WString::new();
+    let mut chars = value.chars();
+    let mut bytes = Vec::new();
+
+    while let Some(c) = chars.next() {
+        let Ok(c) = c else {
+            return Err(Error::AvmError(uri_error(
+                activation,
+                &format!("Error #1052: Invalid URI passed to {func_name} function."),
+                1052,
+            )?));
+		};
```

Comment:
fmt

Reply:
I have `cargo fmt --all`'d this however its still showing this for no reason

Reply:
I've tricked `rustfmt` to format it by making it a `if let` temporarily and that fixed it